### DISCUSSION TITLE
Collapsible folders in "Settings => Data Model"?

### DISCUSSION THREAD

**Discussion by rowild** [directus/directus#14117] [Feature Requests]
[I actually thought that I opened this discussion yesterday already, but I cannot find it. However, if this is a duplication, I apologize and kindly ask you to remove or close it.]

Folders are awesome to structure the content management in the "Content" section. – I was wondering, if there are any plans to inherit this feature in the "Data View" section, too? I do have a lot of tables, mainly junction tables, and the list gets very long. It would be extremely helpful to have collapsible folders there, too.

If this is not on the agenda, would this be something that an extension could add aas a feature? If so, how would that extension roughly work?

**Comment by benhaynes** [comment]
Could you clarify where you mean by "data view"? `Settings > Data Model`?

**Reply by rowild** [reply]
@benhaynes Yes, I meant "Settings -> Data Model" - sorry for being unclear! Thank you for clarifying this! :-)

**Reply by rowild** [reply]
I am thinking of sth very simple, e.g.:

```js
items = document.querySelectorAll('main .v-list > .root-drag-container > .collection-item')

for(let i = 0; i < Array.from(items).length; i++) { 
  items[i].addEventListener('contextmenu', (e) => {

    e.preventDefault()

    const childContainer = items[i].querySelector('.drag-container')

    if(childContainer.style.maxHeight === "0px") {
      childContainer.style.minHeight = null
      childContainer.style.maxHeight = null
      childContainer.style.marginTop = "8px"
    } 
    else {
      childContainer.style.minHeight = 0
      childContainer.style.maxHeight = 0
      childContainer.style.overflow = "hidden"
      childContainer.style.marginTop = 0
    }
  })
}
```

Fooled around in the console and put the event listener on right-click ("contextmenu"), because "click" is already taken. Of course a clean solution with a separate icon on the folder template and an associated vue directive would be more elegant..

**Reply by benhaynes** [reply]
I think an `expand_more` icon on the right (left of the folder) could work, but I defer to @rijkvanzanten on if this is a good idea and regarding the code snippet.

**Reply by rijkvanzanten** [reply]
> I think an expand_more icon on the right (left of the folder) could work

Agreed 👍🏻 

> but I defer to @rijkvanzanten on if this is a good idea and regarding the code snippet.

The code snippet is a nice POC, but not the way we would implement it for real in the app 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
